runtime/util/thrinfo.c

```diff
@@ -287,7 +287,24 @@ getVMThreadStateHelper(J9VMThread *targetThread,
 						 */
 					}
 				}
-	
+
+#if JAVA_SPEC_VERSION >= 19
+				/* For a J9VMThread with a mounted virtual thread that is waiting on a
+				 * lock, the virtual thread object should be returned for the lock object
+				 * and the corresponding J9VMThread should be returned for the lock
+				 * owner.
+				 * This behaviour is to match the RI's JCL behaviour for ThreadMXBean,
+				 * there is no explicit documentation for this. The ThreadMXBean Doc
+				 * specified that VirtualThread are not supported. Hence any lock related
+				 * to VirtualThread has not been defined
+				 */
+				if ((J9VMTHREAD_STATE_WAITING == vmstate) || (J9VMTHREAD_STATE_WAITING_TIMED == vmstate)) {
+					if (NULL != targetThread->currentContinuation) {
```

Comment:
two `if` statements are not needed. they can be combined.
```suggestion
				if ((J9VMTHREAD_STATE_WAITING == vmstate) || (J9VMTHREAD_STATE_WAITING_TIMED == vmstate))
				&& (NULL != targetThread->currentContinuation)
				) {
```